src/miner.cpp

```diff
@@ -648,12 +648,17 @@ bool SignStakeBlock(CBlock &block, CKey &key, vector<const CWalletTx*> &StakeInp
 
 int AddNeuralContractOrVote(const CBlock &blocknew, MiningCPID &bb)
 {
-    std::string sb_contract;
-
     if(OutOfSyncByAge())
         return printf("AddNeuralContractOrVote: Out Of Sync\n");
 
+    if(!IsNeuralNodeParticipant(bb.GRCAddress, blocknew.nTime))
+        return printf("AddNeuralContractOrVote: Not Participating\n");
+
+    if(!NeedASuperblock())
+        return printf("AddNeuralContractOrVote: not Needed\n");
```

Comment:
I understand that "Current" hash should be populated even if Not Needed. In that case, the order of checks was correct and you broke it. But CurrentNeuralHash is only used for stats, so you may consider removing it completely. But not both.

Reply:
Ooh, gotcha. I might revert that entire file then.